src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,188 @@
+                    return true;
+                }
+            }
+        } else {
+            for (Node<T> currentNode = head; currentNode != null; currentNode = currentNode.next) {
+                if (currentNode.value != null && currentNode.value.equals(object)) {
+                    unlink(currentNode);
+                    return true;
+                }
+            }
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    @Override
+    public String toString() {
+        StringBuilder a = new StringBuilder();
+        Node<T> currentNode = head;
+        while (currentNode != null) {
+            a.append(currentNode.value).append(", ");
+            currentNode = currentNode.next;
+        }
+        return a.substring(0, a.length() - 2);
+    }
+
+    private T unlink(Node<T> node) {
+        final T element = node.value;
+        Node<T> next = node.next;
+        Node<T> previous = node.prev;
+
+        if (previous == null) {
+            head = next;
+        } else {
+            previous.next = next;
+            node.prev = null;
+        }
+
+        if (next == null) {
+            tail = previous;
+        } else {
+            next.prev = previous;
+            node.next = null;
+        }
+        node.value = null;
+        size--;
+        return element;
+    }
+
+    private void checkIndexPosition(int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("Index "
+                    + index
+                    + " is out of bounds. Please enter correct index.");
+        }
+    }
+
+    private void checkElementIndex(int index) {
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException("Index "
+                    + index
+                    + " is out of bounds. Please enter correct index.");
+        }
+    }
+
+    private void linkBefore(T value, Node<T> succ) {
+        final MyLinkedList.Node<T> pred = succ.prev;
+        final MyLinkedList.Node<T> newNode = new MyLinkedList.Node<>(pred, value, succ);
+        succ.prev = newNode;
+        if (pred == null) {
+            head = newNode;
+        } else {
+            pred.next = newNode;
+        }
+        size++;
+    }
+
+    private Node<T> getNode(int index) {
+        checkIndexPosition(index);
+        return node(index);
+    }
+
+    private Node<T> node(int index) {
+        if (index < (size / 2)) {
+            Node<T> node = head;
+            for (int i = 0; i < index; i++) {
+                node = node.next;
+            }
+            return node;
+        } else {
+            Node<T> node = tail;
+            for (int i = size - 1; i > index; i--) {
+                node = node.prev;
+            }
+            return node;
+        }
+    }
```

Comment:
I don’t understand why we need the second method, if we can combine it into one. And the very uninformative name "node"